src/main.rs

```diff
@@ -186,6 +186,14 @@ fn main() -> Result<i32> {
         if let Some(auth) = args.auth {
             let (username, password) = parse_auth(auth, url.host_str().unwrap_or("<host>"))?;
             request_builder = request_builder.basic_auth(username, password);
+        } else if !args.ignore_netrc {
+            if let Some(netrc) = read_netrc() {
+                if let Some((username, password)) =
+                    auth_from_netrc(url.host_str().unwrap_or("<host>"), netrc)
```

Comment:
The `"<host>"` up above is for prompting the user, so it's not essential, but here I think we should just skip the netrc if the URL doesn't have a host